### PR TITLE
refactor(router): remove admin v2 intermediate features

### PR DESCRIPTION
**Pull request by sai-harsha-vardhan**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

remove intermediate admin features for v2

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

Sanity admin, payments, captures and refunds for v1 & v2

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by Narayanbhat166 — APPROVED** [review]

File: crates/router/Cargo.toml [hunk 1]
```diff
@@ -127,7 +124,7 @@ x509-parser = "0.16.0"
 # First party crates
 
 analytics = { version = "0.1.0", path = "../analytics", optional = true, default-features = false }
-api_models = { version = "0.1.0", path = "../api_models", features = ["errors"] }
+api_models = { version = "0.1.0", path = "../api_models", features = ["errors"], default-features = false }
```
Comment (SanchithHegde) [inline crates/router/Cargo.toml:127]: Do we need the `default-features = false` here?

File: crates/router/src/core/payment_methods/cards.rs [hunk 2]
```diff
@@ -3738,7 +3732,7 @@ pub async fn call_surcharge_decision_management_for_saved_card(
         .attach_printable("Could not decode the routing algorithm")?
         .unwrap_or_default();
     // TODO: Move to business profile surcharge column
```
Comment (SanchithHegde) [inline crates/router/src/core/payment_methods/cards.rs:3734]: Nit: Could you move this comment down? I believe this was meant to be above the `let surcharge_results = ...` line.

File: crates/router/src/routes/admin.rs [hunk 3]
```diff
@@ -206,10 +206,7 @@ pub async fn delete_merchant_account(
 /// Merchant Connector - Create
```
Comment (SanchithHegde) [inline crates/router/src/routes/admin.rs:206]: Nit: We can remove the doc comments here, since they're anyway being read from the `openapi` crate for the purpose of the OpenAPI documentation. Can be taken up in a separate PR.

File: cypress-tests/package-lock.json
Comment (SanchithHegde) [inline cypress-tests/package-lock.json:1]: Why was this file modified?

**Comment by SanchithHegde** [comment]
Can you also please update `docs/building_docker_images.md` with the updates `docker build` commands?

File: crates/router/src/routes/admin.rs (hunk 3 above)
Comment (sai-harsha-vardhan) [reply crates/router/src/routes/admin.rs:206]: Created an issue for this, will take this up in next PR - https://github.com/juspay/hyperswitch-cloud/issues/6606

**Review by SanchithHegde — APPROVED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

**Review by tsdk02 — APPROVED** [review]

**Review by swangi-kumari — APPROVED** [review]

**Review by ThisIsMani — APPROVED** [review]
Dashboard specific changes looks fine.

**Review by ShankarSinghC — APPROVED** [review]
